src/components/maps/GlobalMap.tsx

```diff
@@ -43,68 +45,74 @@ interface GlobalMapProps {
   }
   onCameraMovement?: (camera: CameraInfo) => void
   children?: React.ReactNode
+  handleOnClick?: (e: MapLayerMouseEvent) => void
+  initialAreaId?: string
 }
 
 /**
  * Global map
  */
 export const GlobalMap: React.FC<GlobalMapProps> = ({
-  showFullscreenControl = true, initialCenter, initialZoom, initialViewState, onCameraMovement, children
+  showFullscreenControl = true, initialCenter, initialZoom, initialViewState, onCameraMovement, children, handleOnClick, initialAreaId
 }) => {
   const [clickInfo, setClickInfo] = useState<ActiveFeature | null>(null)
   const [hoverInfo, setHoverInfo] = useState<ActiveFeature | null>(null)
   const [mapInstance, setMapInstance] = useState<maplibregl.Map | null>(null)
   const [cursor, setCursor] = useState<string>('default')
   const [mapStyle, setMapStyle] = useState<string>(MAP_STYLES.light.style)
+  const [isSourceLoaded, setIsSourceLoaded] = useState(false)
   const [dataLayersDisplayState, setDataLayersDisplayState] = useState<DataLayersDisplayState>({
     areaBoundaries: false,
     organizations: false,
     heatmap: false,
     crags: true
   })
+  const router = useRouter()
+  const urlParams = useUrlParams()
 
   const setActiveFeatureVisual = (feature: ActiveFeature | null, fState: FeatureState): void => {
-    if (feature == null || mapInstance == null) return
+    if (feature === null || mapInstance === null) return
```

Comment:
Can you change triple `===` to double catch both null and undefined?  (there are several instances).  Other than that I'm good to merge.